### ISSUE TITLE
Unknown function params_lookup

### ISSUE THREAD

**Issue by dextervip** [pyrocms/vagrant-pyrocms#4]
Hi,

I cloned devops-vagrant repository, however when i run vagrant up mysql, I am getting the following error:

[mysql] Running provisioner: Vagrant::Provisioners::Puppet...
[mysql] Running Puppet with /tmp/vagrant-puppet/manifests/ubuntu-apache2-mysql-p
hp5.pp...
stdin: is not a tty
Unknown function params_lookup at /tmp/vagrant-puppet/modules-0/php/manifests/in
it.pp:159 on node www

The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

cd /tmp/vagrant-puppet/manifests && FACTER_fqdn='dev.pyrocms-pro.mysql' FACTER_h
ostname='www' FACTER_docroot='/vagrant/www/pyrocms/web' puppet apply --modulepat
h '/tmp/vagrant-puppet/modules-0' /tmp/vagrant-puppet/manifests/ubuntu-apache2-m
ysql-php5.pp

What could be causing it?

**Comment by philsturgeon** [comment]
This was fixed by adding a missing module.

**Comment by Lbatson** [comment]
Still seems to have problems installing. Here's what I'm getting 

[mysql] Running provisioner: puppet...
Running Puppet with ubuntu-apache2-mysql-php5.pp...
stdin: is not a tty
Error: Could not find class apache for www on node www
Error: Could not find class apache for www on node www
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

cd /tmp/vagrant-puppet/manifests && FACTER_fqdn='dev.pyrocms.mysql' FACTER_hostname='www' FACTER_docroot='/vagrant/www/pyrocms/' puppet apply --hiera_config /vagrant/hiera.yaml --modulepath '/etc/puppet/modules:/tmp/vagrant-puppet/modules-0' ubuntu-apache2-mysql-php5.pp --detailed-exitcodes || [ $? -eq 2 ]

Edit: After looking over the git clone --recursive I noticed it's failing to clone and checkout modules/apache. The other submodules work fine and are getting checked out correctly.

Cloning into 'modules/apache'...
remote: Counting objects: 1942, done.
remote: Compressing objects: 100% (1330/1330), done.
remote: Total 1942 (delta 953), reused 1477 (delta 546)
Receiving objects: 100% (1942/1942), 350.64 KiB, done.
Resolving deltas: 100% (953/953), done.
fatal: reference is not a tree: 51dd24029216ccef06a03e5013c5dfbd164b2be7 

Cloning into 'modules/stdlib'...
remote: Counting objects: 3741, done.
remote: Compressing objects: 100% (2063/2063), done.
remote: Total 3741 (delta 1513), reused 3421 (delta 1226)
Receiving objects: 100% (3741/3741), 646.79 KiB, done.
Resolving deltas: 100% (1513/1513), done.
Submodule path 'modules/stdlib': checked out '4e25230dd3899ae117a6b621fc6233cb4b305c5a'
Unable to checkout '51dd24029216ccef06a03e5013c5dfbd164b2be7' in submodule path 'modules/apache'

Edit:

I've updated the submodule for the apache module and sent in a PR to fix it
